marine_benchmark.c

```
@@ -0,0 +1,162 @@
+//
+// Created by reznik on 3/29/20.
+//
+#define PACKET_COUNT 210000
+#define CASES 7
+#define PART_LEN PACKET_COUNT / CASES
+
+#include<stdio.h>
+#include<pcap.h>
+#include "marine.h"
+#include <string.h>
+#include <assert.h>
+#include <time.h>
+#include <stdlib.h>
+#include <sys/resource.h>
+#include <zconf.h>
+
+typedef struct {
+    struct pcap_pkthdr *header;
+    const u_char *data;
+} packet;
+
+
+/*
+* Author:  David Robert Nadeau
+* Site:    http://NadeauSoftware.com/
+* License: Creative Commons Attribution 3.0 Unported License
+*          http://creativecommons.org/licenses/by/3.0/deed.en_US
+*/
+size_t get_current_rss(void) {
+    long rss = 0L;
+    FILE *fp = NULL;
+    if ((fp = fopen("/proc/self/statm", "r")) == NULL) {
+        return (size_t) 0L;
+    }
+    if (fscanf(fp, "%*s%ld", &rss) != 1) {
+        fclose(fp);
+        return (size_t) 0L;
+    }
+    fclose(fp);
+    return (size_t) rss * (size_t) sysconf(_SC_PAGESIZE);
+}
+
+int load_cap(char *file, packet packets[]) {
+    printf("Start loading packets from cap\n");
+
+    char errbuff[PCAP_ERRBUF_SIZE];
+    pcap_t *pcap = pcap_open_offline(file, errbuff);
+    if (pcap == NULL) {
+        printf("Error will opening the cap: %s\n", errbuff);
```

Print errors to `stderr`

---

And the English is pretty funny in the error message

---

Remove the `printf` from this function. The function should return an error code and handling the result will be done by the calling function.